src/testdir/test_ins_complete.vim

```diff
@@ -2586,6 +2586,15 @@ func Test_complete_fuzzy_match()
   call feedkeys("A\<C-X>\<C-N>\<Esc>0", 'tx!')
   call assert_equal('hello help hero h', getline('.'))
 
+  " issue #15095 wrong select when leader is NULL and score both 0.
+  set completeopt=fuzzy,menu
+  call setline(1, ['hello helio hero    h'])
+  call feedkeys("A\<C-X>\<C-P>\<Esc>0", 'tx!')
+  call assert_equal('hello helio hero    hero', getline('.'))
+  call setline(1, ['hello helio hero    h'])
+  call feedkeys("A\<C-X>\<C-P>\<C-P>\<C-P>\<Esc>0", 'tx!')
+  call setline(1, ['hello helio hero    hello'])
```

Comment:
Do you mean to use `assert_equal()` instead of `setline()` here?

Reply:
oops sry wrong paste..